Adds 'Frequently Asked Questions' page

Natolida opened the pull request:
- Adds  FAQ.md to 'Resouces' folder
- Updates 'Index.md' to include

Natolida committed:
Create FAQ.md

Adds FAQ page to resources

Natolida committed:
Update index.md

Adds "Frequently Asked Questions" to Resources Index

Natolida committed:
Merge pull request #1 from Natolida/FAQ

Adds 'Frequently Asked Questions' page
- Adds FAQ.md to resources folder
- Updates Index.md to include 'Frequently Asked Questions'

Natolida commented:
Hey Team,

I think the FAQs page is at a good enough point to add to the docs, and we can continue to build from there. Would you mind reviewing the validity of the information? This is my first significant contribution, and I would hate for any information to be inaccurate. 

Cheers, 
David

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,197 @@
+Cracking of pumps for use with the Trio app is a significant undertaking. At the same time, developers are constantly monitoring pumps that could be integrated into Trio. Ultimately, cracking and creating drivers for these pumps requires multiple developers to devote significant amounts of their time. Realistically, it can take upwards of 12 months for new pump integration. Information on support ...
+
+**I regularly go places without phone service or an internet connection. Will Trio work without these?**
+<br>Yes. Trio does not require internet or cell coverage to work. So long as the Tiro user has Bluetooth enabled on the iPhone, the CGM and insulin pump will continue working in conjunction with Trio. 
```
✏️ Typo: `Tiro` => `Trio`

Natolida replied:
Sorted

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,197 @@
+**How does 'Max SMB Basal Minutes' limit SMBs?**
+<br>`Max SMB Basal Minutes` is one of the major limits on how much insulin a single SMB can deliver. The amount of insulin an SMB can deliver is related to the amount of scheduled basal insulin in your "Basal Profile" settings.
+
+Example: John is using Trio and has SMBs turned on in the settings. Johns's current basal rate is if you are receiving a 1 unit per hour. John has `Max SMB Basal minutes` set to the default of 30. Therefore, a single SMB can deliver no more insulin than would be delivered by your basal in 30 minutes, i.e., 0.5 units. 
```
✏️ Typo:  `Johns's` => `John's`

Natolida replied:
Sorted. Thank you :)

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,197 @@
+**How does 'Max SMB Basal Minutes' limit SMBs?**
+<br>`Max SMB Basal Minutes` is one of the major limits on how much insulin a single SMB can deliver. The amount of insulin an SMB can deliver is related to the amount of scheduled basal insulin in your "Basal Profile" settings.
+
+Example: John is using Trio and has SMBs turned on in the settings. Johns's current basal rate is if you are receiving a 1 unit per hour. John has `Max SMB Basal minutes` set to the default of 30. Therefore, a single SMB can deliver no more insulin than would be delivered by your basal in 30 minutes, i.e., 0.5 units. 
```
✏️ Typo: If I understood correctly, `if you are receiving a` seems unnecessary in this sentence.

In this case, I suggest replacing the following:
> Johns's current basal rate is if you are receiving a 1 unit per hour.

with
> John's current basal rate is 1 unit per hour.

Natolida replied:
Sorted

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,197 @@
+Example: John has SMBs and UAM turned on in the settings. The algorithm has detected unexpected carb absorption and would like to deliver an SMB in response. Johns's current basal rate is if you are receiving a 1 unit per hour. John has `Max UAM SMB Basal minutes` set to the default of 30. Therefore, a single SMB can deliver no more insulin than would be delivered by your basal in 30 minutes, i.e ...
+
+**I've turned on SMBs, but every time I receive an SMB, Trio cuts back or zero-temps my basal. Is that normal?**
+<br>Yes. SMBs effectively borrows insulin from the future to front load insulin delivery to assist in avoiding glucose spikes. Long, low, or zero temporary basal rates are expected to be observed with SMBs activated. This allows the algorithm to balance out the peak insulin timing and avoid hypoglycemia secondary to the increased insulin on board. 
```
💡 TIL! Thanks.

Natolida replied:
Sorted. Thank you!

ebouchut commented on docs/EN/resources/FAQ.md:
I suggest using a level-3 heading for each FAQ sub-item instead of bold.
That is, for instance, replacing:
```markdown
**What CGMs are supported by Trio?**
```
with
```markdown
### What CGMs are supported by Trio?
```

Natolida replied:
I did have that initially. Although, it made the index quite long. I have changed it back to include the mackdown  headings. Search-ability is more important :)

ebouchut reviewed (commented):
I found a couple of typos (✏️).   
I also suggest using Markdown headings for FAQ articles to improve their searchability and navigability.

Thank you @Natolida, as a new Trio user, I found this page useful.

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,197 @@
+<br>Trio can’t determine the glucose value when under 40mg/dL (2.2 mmol/L). This is why the CGM apps from Dexcom and Libre are displaying “Low.” A “Low” reading is also sometimes due to a sensor error or a faulty sensor needing replacement. Trio can't safely make a glucose prediction without knowing the glucose value or if it’s rising or falling.
+
+**Why isn't Trio looping or delivering insulin when CGM glucose is 400 mg/dl (22 mmol/l) or higher?**
+<br>Trio can’t determine the glucose value when over 400mg/dL (22 mmol/L). This is why the CGM apps from Dexcom and Libre display “High.” A “High” reading is also sometimes due to a sensor error or a faulty sensor needing replacement. Trio can't safely make a glucose prediction without knowing the glucose value or if it’s rising or falling.
```
ℹ️ **TIP** 
To add a line break after a sentence, instead of using the `<br>` HTML tag, you can **add two spaces immediately after the last word of the sentence**. 
That is, use
> ```markdown
> [...] or if it’s rising or falling.␠␠
> ```
> ℹ️ In the above example, I used `␠` to **visually** denote and materialize a space character (` `).

instead of:
> ```markdown
> [...] or if it’s rising or falling.
> <br>
> ```

Natolida replied:
Thanks for the tip.

Natolida committed:
Update FAQ.md

- Corrects typo's
- Removes erroneous wording
- Adds markdown headings to questions
- Limits used of <br> instead using double space

ebouchut commented on docs/EN/resources/FAQ.md:
```diff
@@ -81,19 +81,19 @@ _All options for following your Trio app rely on an active internet connection.
 
-**Why does the 'Recommended Bolus Percentage' default to 70%?**
-<br>`Recommended Bolus Percentage` is a safety feature built into Trio. By default, Trio first calculates an "insulin recommended" value when bolusing for carbs, which is the total dosage. That dosage is multiplied by your `Recommended Bolus Percentage` to display your suggested insulin dose. Trio then delivers the remaining insulin via Temp Basals and/or SMBs as the blood sugar rises.
+###Why does the 'Recommended Bolus Percentage' default to 70%?
```
✏️ There is a space missing between the hash signs and the first word.

Natolida replied:
Fixed, thank you!

ebouchut reviewed (commented):
I found one typo, otherwise 👍 LGTM regarding the structure.

Natolida committed:
Update FAQ.md

- Fixes typo
- Name change from John to Bill in examples for continuity through docs.

Natolida commented:
Hey @tmhastings, I am wondering if it would be a good idea to get this in prior to the docs migration?

tmhastings commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,195 @@
+* The 'Mac Build' method using XCode.
+
+### Do I need to own an Apple computer?
+You no longer need to own an Apple computer. If you chose the 'Mac Build' method, you still don't have to own an Apple computer, but you must at least borrow one.  
```
Let's add a statement here that clarifies a mac is not needed for Browser Build through GitHub.

Natolida replied:
All sorted

tmhastings commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,195 @@
+
+### Does Trio cause an increase in pod failures?
+There is more communication between your phone and the pod than the PDM. This increases the load on the pod battery. Most people have no increase in pod failures. However, it does occasionally happen. Below are some tips to take to limit battery drain:
+* Have a correction range instead of a single number
```
This is only applicable to Loop, not Trio. Can be removed from FAQs.

Natolida replied:
Fixed

tmhastings commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,195 @@
+### Does Trio cause an increase in pod failures?
+There is more communication between your phone and the pod than the PDM. This increases the load on the pod battery. Most people have no increase in pod failures. However, it does occasionally happen. Below are some tips to take to limit battery drain:
+* Have a correction range instead of a single number
+* The pod state is updated every 3 minutes for DASH and every 5 minutes for Eros. If uncertain communications are detected such that the Loop app cannot determine if a dosing command sent to the pod was received, the app will try once a minute to get a response.
```
Replace "the Loop app" with "Trio"

Natolida replied:
All sorted

tmhastings commented on docs/EN/resources/FAQ.md:
```diff
@@ -0,0 +1,195 @@
+:::
+
+**Nightscout**
+<br>Yes. Nightscout can be used as a remote CGM source in Trio. However, this option should be avoided because it does not keep Trio running in the background like other CGM options. You will have to open Trio manually to make it run loop cycles.  
```
I believe this has been fixed in Trio 0.2.0's Nightscout integration update, but need @MikePlante1 to confirm.

Natolida replied:
I have left this in, for now; the information is accurate per the current Trio docs. If you are happy to merge, I can update it in a future PR along with the source of the information.

tmhastings reviewed (commented):
Left a few comments on places that need some minor edits. Once fixed, I'm ready to approve. Thank you for putting this together. It looks great!

Natolida committed:
Update FAQ.md

- Fixes Typo
- Adopts suggested changes as per T. Hastings

Natolida commented:
Thanks team, all updated and should be ready to go.

Natolida commented:
Was there anything else I needed to do for a merge?

ebouchut commented:
@Natolida 
Our main branch, `upstream/dev`, has seen a lot of changes recently. This is because we switched to `Mkdocs` as our website generation engine.  
This is why, we first need to integrate them into your topic/feature/PR branch which is ... `dev`.
Then we need to update the Table of Contents (TOC) which has moved to `mkdocs.yml` from `docs/index.html`.
Finally, we also need to fix links that are now broken due to a change in the folder structure.

The PR branch is not supposed to be `dev', but don't worry, I'll explain how to fix that.



## Merge Changes from Upstream

Integrate the recent changes from the `upstream/dev` branch locally into your `dev` branch.
```shell
cd trio-docs
source venv/bin/activate 
    
# Fetch the recent changes from the `upstream` repository (`nightscout/trio-docs`)
# - `origin` your remote clone repository (i.e. `Natolida/trio-docs`)
# and store them locally in the remote cache
git fetch upstream

# Fetch the recent changes from the `origin` repository (your remote clone repository: `Natolida/trio-docs`)
# and store them locally in the remote cache
# -p means prune, that is if remove branches from the remote `origin` cache that have been removed from `Natolida/trio-docs`
git fetch -p origin 

# Jump on your topic/feature/PR branch which is ... `dev`
git switch dev


# Rebase `dev` onto `upstream/dev`, that is replay the commits from `dev` on top of the `upstream/dev` branch.
# as if we tack your `dev` branch on top of `upstream/dev`
git rebase upstream/dev

# You will then get a conflict because Git noticed that
# the `docs/resources/FAQ.md` file exist in your branch 
# but not in the other.
# Git stopped rebasing to ask you to choose which one to keep.

# Display the current state of rebase and conflicts
git status

# Choose yours version of the file 
git checkout --theirs -- docs/resources/FAQ.md

# I confirm: `--theirs` above denotes your file (on `dev`).
#
# When starting a `git rebase upstream/dev`:
# - Git jumps on `upstream/dev` wich becomes `HEAD` (FYI `HEAD` the tip of the current branch)
# - `--ours` always denotes `HEAD`(`upstream/dev` in this case)
# - `--theirs` denotes `dev` (your `dev` branch that is the one you were on when you ran `git rebase`)
#
# It is a bit far-fetched!
# It's just the opposite of what you expect.
# Linus probably expected someone to add software layers on top of its initial implementation of Git
# to make this more user friendly and consistent.
#
# The rule of thumb is to find where `HEAD` points to.
# HEAD denotes `--ours` 
# During a `git rebase`:
# - `--ours`    ==> the branch you are rebasing (`upstream/dev`)
# - `--theirs` ==> the branch where you ran `git rebase` from (`dev`)

# To complicate things, it is the exact opposite for `git merge upstream/dev`
# See `git rebase` section here for a more visual explanation:
#   https://nitaym.github.io/ourstheirs/ 

# Inform Git that we resolved the conflict
git add docs/resources/FAQ.md

# Ask Git to continue Rebasing
git rebase --continue

# Git opens up an editor with the merge commit message.
# It is the same as your last commit.
# Left it as is, save and exit the editor

# Git continue rebasing the next commit
# There is another conflict. This time it is `docs/index.md`
git status # to have a look at which file is in conflict

# In this commit you added FAQS.md to the Table of Contents (TOC) which was in this file 
# This was back when the site used Sphinx as its website engine.
# Since then we have switched to Mkdocs and the TOC is now located in `mkdocs.yml` blow the `nav:` section.
# We need to:
# - A) Remove the changes you made to `docs/index.md`
# - B) Add FAQS.md to the Table of Contents in `mkdocs.md`

# A) Ask git to forget about your changes to `docs/index.md` and use the ones from `dev/upstream`
git checkout --ours -- docs/index.md

# B) Add FAQS.md to the `nav:`section in `mkdocs.yml`
#    Edit `mkdocs.yml` 
#    Append this line to this section (at the bottom of the TOC)
#      - FAQ: ./resources/FAQ.md

YOUR_FAVORITE_EDITOR_HERE mkdocs.yml

# Inform Git that we have fixed the conflicts
git add mkdocs.yml docs/index.md

# Ask Git to continue rebasing
git rebase --continue

# Once asked, update the commit message like so:
#   Add FAQS to the Table of Contents
# Save and exit the editor.

# Congratulations, you are done rebasing!


# Remove the currently installed packages (i.e. dependencies) to start from a blank slate
python -m pip freeze --exclude-editable | xargs python -m pip uninstall -y

# Install the packages
pip install -r dev-requirements.txt
pip install -r requirements.txt

# You can now live preview the changes you make to the TrioDocs localy
# See: https://github.com/nightscout/trio-docs?tab=readme-ov-file#preview-changes
mkdocs serve

# You now need to fix the links in the FAQ page as the directory structure has changed.
# The links in this page are now broken because `EN/latest/` has been dropped since the conversion to Mkdocs.
# You need to update them, commit a fix and push the branch 

YOUR_FAVORITE_EDITOR_HERE docs/resources/FAQ.md
git commit -m "Fix broken links according to new folder structure"
git push origin 

``


## Fix dev Branch

IMPORTANT: Do the following once but ONLY AFTER PR #81 has been merged

The `dev` branch should not be used to hold the commits of your topic/feature.
To fix this the soft way, we will **wait for the PR to be merged first**.
Then we will move your `dev` branch so that it points to `upstream/dev`, like this:

```shell
cd trio-docs
source venv/bin/activate 
    
git fetch upstream
git fetch origin

git switch dev

# IMPORTANT: Make sure that you do not have files that are not committed (i.e. not under version control).
# If so you here is how hto stash your work in progress. 
# We will set these files aside on a plate with the tag "MyWIP", in a special drawer that Git calls the "stash". 
# We push this plate on top of the stack of plates already present in this drawer.
git stash push --include-untracked -m MyWIP

# In the future and on another branch, once you need to get these files 
# back into your working directory, ask git to retrieve the plate with the tag "MyWIP"
## git stash pop MyWIP

# Now we need to restore `dev` so that it points at the same commit as `upstream/dev`
# and make some cleaning in your working directory and staging area:
# - Restore a clean working directory
# - Empty the staging area
# - update the tip of the curent branch (`dev`) so that it points to `upstream/dev`
git reset --hard upstream/dev

# Send this branch to your clone repository (`origin` => `Natolida/trio-docs`)
git push --force origin dev
```


## How to create a Pull-Request

- open the your clone repository (`Natolida/trio-docs`) on GitHub
  https://github.com/Natolida/trio-docs
- Click the `Pull requests` (PR) tab at the top
- You should see a yellow box asking if you want to create a PR from your topic branch
- Click this button and fill out the PR creation form

If you do not see the yellow box to create the PR:
- Click the `Code` tab
- Click the `NNN Branches` link next to `Tabs` below the repository name 
- A page with list of existing branches opens up
- Click the triple dots next to the line with your branch 
- Click `New Pull Request` in the drop-down menu
- Fill-out the PR creation form

GH Doc: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request#creating-the-pull-request

Natolida commented:
Hey @ebouchut, I ran into some difficulties with your instructions above. To make things simple, I am going to close this PR. I will update my repo with the changes that have been made recently. I will then put my FAQ work back into its own branch and resubmit the PR.
